contracts/strategies/direct-grants-simple/DirectGrantsSimpleStrategy.sol

```diff
@@ -0,0 +1,388 @@
+        Recipient memory recipient = _getRecipient(_recipientId);
+        if (recipient.recipientStatus == InternalRecipientStatus.Accepted) {
+            return RecipientStatus.Accepted;
+        } else if (recipient.recipientStatus == InternalRecipientStatus.Rejected) {
+            return RecipientStatus.Rejected;
+        } else if (recipient.recipientStatus == InternalRecipientStatus.None) {
+            return RecipientStatus.None;
+        } else {
+            return RecipientStatus.Pending;
+        }
+    }
+
+    /// @notice Returns the payout summary for the accepted recipient
+    function getPayouts(address[] memory _recipientIds, bytes memory, address)
+        external
+        view
+        returns (PayoutSummary[] memory payouts)
+    {
+        uint256 recipientLength = _recipientIds.length;
+
+        payouts = new PayoutSummary[](recipientLength);
+
+        for (uint256 i = 0; i < recipientLength;) {
+            payouts[i] = PayoutSummary(_recipientIds[i], _recipients[_recipientIds[i]].grantAmount);
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Checks if address is elgible allocator
+    /// @param _allocator Address of the allocator
+    function isValidAllocator(address _allocator) external view returns (bool) {
+        return allo.isPoolManager(poolId, _allocator);
+    }
+
+    /// @notice Get the status of the milestone of an recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _milestoneId Id of the milestone
+    function getMilestoneStatus(address _recipientId, uint256 _milestoneId) public view returns (RecipientStatus) {
+        return milestones[_recipientId][_milestoneId].milestoneStatus;
+    }
+
+    /// ===============================
+    /// ======= External/Custom =======
+    /// ===============================
+
+    /// @notice Set milestones for recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _milestones The milestones to be set
+    function setMilestones(address _recipientId, Milestone[] memory _milestones) external onlyPoolManager(msg.sender) {
+        if (upcomingMilestone[_recipientId] != 0) {
+            revert MILESTONES_ALREADY_SET();
+        }
+
+        uint256 totalAmountPercentage;
+
+        uint256 milestonesLength = _milestones.length;
+        for (uint256 i = 0; i < milestonesLength;) {
+            totalAmountPercentage += _milestones[i].amountPercentage;
+            milestones[_recipientId].push(_milestones[i]);
+
+            unchecked {
+                i++;
+            }
+        }
+
+        if (totalAmountPercentage != 1e18) {
+            revert INVALID_MILESTONE();
+        }
+
+        emit MILESTONES_SET(_recipientId);
+    }
+
+    /// @notice Submit milestone by the recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _metadata The proof of work
+    function submitMilestone(address _recipientId, Metadata calldata _metadata) external {
```

Comment:
should we let the user pass the milestoneId he wants to set?

pro: user can submit upcoming milstones without waiting for the next payout. Could be interesting if they work faster than expected

Reply:
Makes sense to add it in